clippy_lints/src/temporary_dropped_with_sideeffects.rs

```diff
@@ -0,0 +1,343 @@
+use super::utils;
+use super::utils::match_def_path;
+use super::utils::paths::*;
+use rustc::hir::map::Node;
+use rustc::hir::{def_id::DefId, Expr, ExprKind, Item, ItemKind, Pat, PatKind};
+use rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use rustc::ty::{subst::UnpackedKind, TyCtxt, TyS, TypeVariants};
+use rustc::{declare_lint, lint_array};
+
+/// **What it does:**
+/// Checks for temporaries containing values where dropping has a side effect.
+///
```

Comment:
I think it would be good to include a list of links to the stdlib to the different types this lint checks.